### PR TITLE
CI on Piz Daint

### PR DESCRIPTION
**Pull request by lukasgd**
Opening this pull request to discuss the development of CI on Piz Daint.

### CODE REVIEW

File: docker/Dockerfile_cscs_ci.deploy.gpu [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM lukasgd/korali_cscs_ci_gpu:base
+
+RUN git clone https://github.com/cselab/korali.git
+RUN cd korali; CC=gcc CXX=g++ meson setup build --prefix=/usr/local --buildtype=release -Donednn=true -Dmpi=true
```
Comment (lukasgd) [inline docker/Dockerfile_cscs_ci.deploy.gpu:4]: The build command probably needs to be adapted for GPU support (cuDNN, etc. in base image).

File: docker/Dockerfile_cscs_ci.deploy.cpu [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM lukasgd/korali_cscs_ci_cpu:base
+
+RUN git clone https://github.com/cselab/korali.git
+RUN cd korali; CC=gcc CXX=g++ meson setup build --prefix=/usr/local --buildtype=release -Donednn=true -Dmpi=true
```
Comment (lukasgd) [inline docker/Dockerfile_cscs_ci.deploy.cpu:4]: Need to double-check if we're compiling for the right CPU architecture here (compilation is not run on Piz Daint itself).

**Comment by lukasgd** [comment]
There is a problem with initialization of MPICH in Korali in the cmaes test example, cf. [here](https://gitlab.com/cscs-ci/ci-testing/webhook-ci/korali-5865265036453491/-/jobs/1872596868#L1221) and reproduced on the command line ([run-cmaes.log](https://github.com/cselab/korali/files/7694454/run-cmaes.log)).

It can't be reproduced by a simple MPI test with the container built with Korali, e.g.
[mpitutorial.log](https://github.com/cselab/korali/files/7694433/mpitutorial.log).

This is either an issue in Korali's code or the way it's built in the current Dockerfile.

**Comment by lukasgd** [comment]
A further improvement to the current CI setup would be to build the tests beforehand (in the build stage) instead of on Daint (test stage) as now.

File: docker/Dockerfile_cscs_ci.deploy.gpu (hunk 1 above)
Comment (wadaniel) [reply docker/Dockerfile_cscs_ci.deploy.gpu:6]: although our installation allows compilation with cuDNN, it is hardly tested and used. we think of deprecating it. so this shall be fine .

File: docker/Dockerfile_cscs_ci.deploy.cpu (hunk 2 above)
Comment (wadaniel) [reply docker/Dockerfile_cscs_ci.deploy.cpu:6]: whats your status on that?

Comment (bcumming) [reply docker/Dockerfile_cscs_ci.deploy.cpu:6]: We are not compiling on Daint at the moment.
When we move the build system to Alps we can think about adding target-specific flags.

**Review by wadaniel — APPROVED** [review]

**Review by YiannisKalogeris — APPROVED** [review]

**Comment by wadaniel** [comment]
I have to close and reopen this PR. I had to update the circle ci settings, such that we allow to run the CI build for PRs from forked projects. Unfortunately this doesn't apply to PRs created before changing the settings.

**Pull request closed without merging**